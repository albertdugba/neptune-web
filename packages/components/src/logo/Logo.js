import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Logo.css';

const Type = {
  WISE: 'WISE',
  WISE_BUSINESS: 'WISE_BUSINESS',
};

const baseUrl = 'https://wise.com/public-resources/assets/logos/wise/';

const logoPaths = {
  WISE: 'brand_logo.svg',
  WISE_BUSINESS: 'brand_logo_business.svg',
  WISE_INVERSE: 'brand_logo_inverse.svg',
  WISE_BUSINESS_INVERSE: 'brand_logo_business_inverse.svg',
};

const Logo = ({ className, inverse, type }) => {
  const path = logoPaths[`${type}${inverse ? '_INVERSE' : ''}`];
  return (
    <img
      className={classNames('np-logo', className)}
      alt={type === Type.WISE ? 'Wise' : 'Wise business'}
      src={`${baseUrl}${path}`}
    />
  );
};

Logo.propTypes = {
  /** Extra classes applied to Logo */
  className: PropTypes.string,
  /** If true, will use dark colours for dark on light theme */
  inverse: PropTypes.bool,
  /** What type of logo to display */
  type: PropTypes.oneOf(['WISE', 'WISE_BUSINESS']),
};

Logo.defaultProps = {
  className: undefined,
  inverse: false,
  type: Type.WISE,
};

export default Logo;

export { Type as LogoType };
