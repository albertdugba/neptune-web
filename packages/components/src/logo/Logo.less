.np-logo {
  height: var(--size-24);
}
