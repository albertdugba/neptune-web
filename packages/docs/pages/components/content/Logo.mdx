import { Logo, LogoType } from '@transferwise/components';

import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
import code from '../../../liveEditorCode/logo.code';

Renders a Wise or Wise Business logo in normal or inverse.

<div>
  <LiveEditorBlock code={code} scope={{ Logo, LogoType }} />
  <GeneratePropsTable componentName="Logo" />
</div>

export const meta = {
  name: 'Logo',
  badge: {
    type: 'new',
    expiryDate: '2021-04-30',
  },
};
