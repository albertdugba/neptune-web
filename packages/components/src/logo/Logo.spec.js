import { render } from '../test-utils';
import Logo, { LogoType } from '.';

describe('Logo', () => {
  it('renders wise logo', () => {
    expect(render(<Logo className="class" type={LogoType.WISE} />).container).toMatchSnapshot();
  });

  it('renders wise logo by default', () => {
    expect(render(<Logo className="class" />).container).toMatchSnapshot();
  });

  it('renders wise business logo', () => {
    expect(
      render(<Logo className="class" type={LogoType.WISE_BUSINESS} />).container,
    ).toMatchSnapshot();
  });

  it('renders wise logo inversed', () => {
    expect(
      render(<Logo className="class" type={LogoType.WISE} inverse />).container,
    ).toMatchSnapshot();
  });

  it('renders wise business logo inversed', () => {
    expect(
      render(<Logo className="class" type={LogoType.WISE_BUSINESS} inverse />).container,
    ).toMatchSnapshot();
  });
});
