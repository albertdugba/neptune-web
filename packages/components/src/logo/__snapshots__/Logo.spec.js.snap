// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders wise business logo 1`] = `
<div>
  <img
    alt="Wise business"
    class="np-logo class"
    src="https://wise.com/public-resources/assets/logos/wise/brand_logo_business.svg"
  />
</div>
`;

exports[`Logo renders wise business logo inversed 1`] = `
<div>
  <img
    alt="Wise business"
    class="np-logo class"
    src="https://wise.com/public-resources/assets/logos/wise/brand_logo_business_inverse.svg"
  />
</div>
`;

exports[`Logo renders wise logo 1`] = `
<div>
  <img
    alt="Wise"
    class="np-logo class"
    src="https://wise.com/public-resources/assets/logos/wise/brand_logo.svg"
  />
</div>
`;

exports[`Logo renders wise logo by default 1`] = `
<div>
  <img
    alt="Wise"
    class="np-logo class"
    src="https://wise.com/public-resources/assets/logos/wise/brand_logo.svg"
  />
</div>
`;

exports[`Logo renders wise logo inversed 1`] = `
<div>
  <img
    alt="Wise"
    class="np-logo class"
    src="https://wise.com/public-resources/assets/logos/wise/brand_logo_inverse.svg"
  />
</div>
`;

exports[`Logo on mobile renders only fast flag 1`] = `
<div>
  <img
    alt="Wise"
    class="np-logo class"
    src="https://wise.com/public-resources/assets/logos/wise/brand_flag.svg"
  />
</div>
`;
